lib/activedirectory/core/locate.py

```diff
@@ -56,13 +57,14 @@ class Locator(object):
     _maxservers = 3
     _timeout = 300  # cache entries for 5 minutes
 
-    def __init__(self, site=None):
+    def __init__(self, site=None, resolver=None):
```

Comment:
Is [here](https://github.com/theatlantic/Atlantic-CMS/blob/develop/apps/amc_ldap/auth_backends.py#L143) where we'd want to set the custom resolver in Ollie? Relatedly, I looked at the default resolver and it looks..rather involved. Would the idea be to subclass it and simply replace the `nameservers` property with ours (and specifically, the hardcoded ips)?

Reply:
Yes. Here's the updated version of that method:

```python
@method_decorator(sensitive_variables('password'))
def get_ad_client(self, username=None, password=None):
    if not self.settings.DOMAIN:
        return None

    if not username and not password:
        username = self.settings.BIND_USER
        password = self.settings.BIND_PASSWORD

    domain = '.'.join(self.settings.DOMAIN).upper()
    creds = activedirectory.Creds(domain)
    locator = activedirectory.core.object.factory(activedirectory.Locator)
    if self.settings.NAMESERVERS:
        locator.m_resolver = dns.resolver.Resolver()
        locator.m_resolver.nameservers = self.settings.NAMESERVERS
    if self.settings.ACTIVE_DIRECTORY_SITE_NAME:
        locator.m_site = self.settings.ACTIVE_DIRECTORY_SITE_NAME
        locator.m_site_detected = True
    creds._resolve_servers_for_domain(creds.m_domain)
    creds.acquire(username, password)
    activedirectory.activate(creds)
    return activedirectory.Client(domain, creds)
```

Once this is merged I'll open a separate pull request against ollie.